memiavl/db.go

```diff
@@ -97,13 +98,28 @@ func (db *DB) Commit(changeSets MultiChangeSet) ([]byte, int64, error) {
 			}
 
 			// snapshot rewrite succeeded, catchup and switch
-			// TODO prune the old snapshots
 			if err := result.mtree.CatchupWAL(db.wal); err != nil {
 				return nil, 0, fmt.Errorf("catchup failed: %w", err)
 			}
 			if err := db.reloadMultiTree(result.mtree); err != nil {
 				return nil, 0, fmt.Errorf("switch multitree failed: %w", err)
 			}
+			// prune the old snapshots
+			version := uint32(db.lastCommitInfo.Version)
+			latestSnapshot := snapshotName(version)
+			prefix := strings.TrimSuffix(latestSnapshot, fmt.Sprintf("%d", version))
```

Comment:
The prefix should be just a constant, "snapshot-"

Reply:
it's const after trim, do you mean hardcode to `snapshot-`?

Reply:
yes